File: php/problem9.php
<?php
//solution: 31875000
echo eulerProblem9() . "\n";

function eulerProblem9() {
   for($a = 1; $a < 1000; $a++) {
       for($b = 1; $b < 1000; $b++) {
           if(1000 - ($a + $b) < 1) {
               continue;
           }
           
           $c = 1000 - ($a + $b);
           
           if($a * $a + $b * $b == $c * $c) {
               return $a * $b * $c;
           }
       }
   }
   
   return -1; //should never happen
}

?>

File: cpp/problem9.cpp
#include <iostream>
using namespace std;
//solution: 31875000

int eulerProblem9() {
    for(int i = 1;; ++i) {
        for(int j = i+1; j+i < 1000; ++j) {
            int k = 1000 - j - i;
            if(k*k == j*j + i*i) {
                return i*j*k;
            }
        }
    }
}

int main() {
    cout << eulerProblem9() << endl;
    return 0;
}
